File: src/Domain/Exception/RoleAlreadyExistsException.php
<?php

declare(strict_types=1);

namespace App\Domain\Exception;

class RoleAlreadyExistsException extends DomainException
{
    public function __construct(string $message = 'Já existe um role com o nome informado.')
    {
        parent::__construct($message, 409);
    }

    public function getErrorCode(): string
    {
        return 'ROLE_ALREADY_EXISTS';
    }

    public function getHttpStatusCode(): int
    {
        return 409;
    }
}

File: src/main/java/br/pucrs/constrsw/oauth/domain/exception/RoleAlreadyExistsException.java
package br.pucrs.constrsw.oauth.domain.exception;

/** Ja existe role com esse nome. Mapeada em HTTP 409. */
public class RoleAlreadyExistsException extends DomainException {

    public RoleAlreadyExistsException(String name) {
        super("Role already exists: " + name);
    }
}
